lib/active_model/serializer/type.rb

```diff
@@ -16,8 +16,17 @@ module ClassMethods
         # @example
         #   class AdminAuthorSerializer < ActiveModel::Serializer
         #     type 'authors'
-        def type(type)
-          self._type = type && type.to_s
+        #
+        # @example
+        #   class AdminAuthorSerializer < ActiveModel::Serializer
+        #     type { |object| object.class.name }
+        #
+        # @example
+        #   class AdminAuthorSerializer < ActiveModel::Serializer
+        #     type proc { |object| object.class.name }
+        #
+        def type(type = nil, &block)
+          self._type = block || type
```

Comment:
> also, how does this not blow up the serializer's json_key?
 
maybe we having a missing test?

> I mean, it's in return serializer._type.to_s if serializer._type below, but why not just do the conditional once :)

yeah, idk .I could do `block || type.try(:to_s)` to compactly achieve the same, I think